google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolOptions.java

```diff
@@ -369,6 +545,70 @@ public Builder setBlockIfPoolExhausted() {
       return this;
     }
 
+    /**
+     * If there are inactive transactions, log warning messages with the origin of such transactions
+     * to aid debugging. A transaction is classified as inactive if it executes for more than a
+     * system defined duration.
+     *
+     * <p>This option won't change the state of the transactions. It only generates warning logs
+     * that can be used for debugging.
+     *
+     * @return this builder for chaining
+     */
+    Builder setWarnIfInactiveTransactions() {
+      this.inactiveTransactionRemovalOptions =
+          InactiveTransactionRemovalOptions.newBuilder()
+              .setActionOnInactiveTransaction(ActionOnInactiveTransaction.WARN)
+              .build();
+      return this;
+    }
+
+    /**
+     * If there are inactive transactions, release the resources consumed by such transactions. A
+     * transaction is classified as inactive if it executes for more than a system defined duration.
+     * The option would also produce necessary warning logs through which it can be debugged as to
+     * what resources were released due to this option.
+     *
+     * <p>Use the option {@link Builder#setWarnIfInactiveTransactions()} if you only want to log
+     * warnings about long-running transactions.
+     *
+     * @return this builder for chaining
+     */
+    Builder setWarnAndCloseIfInactiveTransactions() {
```

Comment:
why are we making these methods package protected?

Reply:
@psinghbay1 This feature is not yet launched. Hence it is package protected in this PR. We will be enabling this in the next couple of weeks.